api/v1alpha1/ext_proc_types.go

```diff
@@ -0,0 +1,29 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+// +kubebuilder:validation:XValidation:rule="has(self.service) ? (!has(self.service.backendRef.group) || self.service.backendRef.group == \"\") : true", message="group is invalid, only the core API group (specified by omitting the group field or setting it to an empty string) is supported"
+// +kubebuilder:validation:XValidation:rule="has(self.service) ? (!has(self.service.backendRef.kind) || self.service.backendRef.kind == 'Service') : true", message="kind is invalid, only Service (specified by omitting the kind field or setting it to 'Service') is supported"
+//
+// ExtProc defines the configuration for External Processing filter.
+type ExtProc struct {
+	// Service defines the configuration of the external processing service
+	Service ExtProcService `json:"service"`
```

Comment:
can we directly use `BackendRef` here ?

Reply:
There are some additional backend-related attributes that we may want to support here in the future. For example, passing initial metadata to authenticate with the external-processing backend, choosing the sort of grpc client (envoy vs. google), etc. In Ext-Auth, we decided to use the a "service" container for this purpose: https://github.com/envoyproxy/gateway/blob/d020ce742bf873052bce590fcb3fb0a73e71ee77/api/v1alpha1/ext_auth_types.go#L66

Reply:
if we want to leave room for expansion, thoughts on ?
```
type ExtProc struct {
......
BackendRef BackendRef 
```

type BackendRef struct {
gwapiv1.BackendObjectReference
.....
}

Reply:
it matches the style adopted in upstream https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1.BackendRef where additional fields such as `weight` are added